atc/exec/get_step.go

```diff
@@ -67,48 +65,27 @@ func NewGetStep(
 	plan atc.GetPlan,
 	metadata StepMetadata,
 	containerMetadata db.ContainerMetadata,
-	resourceFetcher fetcher.Fetcher,
+	resourceFactory resource.ResourceFactory,
 	resourceCacheFactory db.ResourceCacheFactory,
 	strategy worker.ContainerPlacementStrategy,
 	workerPool worker.Pool,
 	delegate GetDelegate,
+	client worker.Client,
 ) Step {
 	return &GetStep{
 		planID:               planID,
 		plan:                 plan,
 		metadata:             metadata,
 		containerMetadata:    containerMetadata,
-		resourceFetcher:      resourceFetcher,
+		resourceFactory:      resourceFactory,
 		resourceCacheFactory: resourceCacheFactory,
 		strategy:             strategy,
 		workerPool:           workerPool,
 		delegate:             delegate,
+		workerClient:         client,
 	}
 }
 
-// Run ultimately registers the configured resource version's ArtifactSource
-// under the configured SourceName. How it actually does this is determined by
-// a few factors.
-//
-// First, a worker that supports the given resource type is chosen, and a
-// container is created on the worker.
-//
-// If the worker has a VolumeManager, and its cache is already warmed, the
-// cache will be mounted into the container, and no fetching will be performed.
-// The container will be used to stream the contents of the cache to later
-// steps that require the artifact but are running on a worker that does not
-// have the cache.
-//
-// If the worker does not have a VolumeManager, or if the worker does have a
-// VolumeManager but a cache for the version of the resource is not present,
-// the specified version of the resource will be fetched. As long as running
-// the fetch script works, Run will return nil regardless of its exit status.
-//
-// If the worker has a VolumeManager but did not have the cache initially, the
-// fetched ArtifactSource is initialized, thus warming the worker's cache.
-//
-// At the end, the resulting ArtifactSource (either from using the cache or
-// fetching the resource) is registered under the step's SourceName.
```

Comment:
We should make a note to come back here and add better description of what's happening, once this refactor is finished. Definitely good to remove it for now since incorrect documentation is worse than no documentation.